dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedNodeSelectionStrategyChannel.java

```diff
@@ -58,47 +47,19 @@
 final class BalancedNodeSelectionStrategyChannel implements LimitedChannel {
     private static final Logger log = LoggerFactory.getLogger(BalancedNodeSelectionStrategyChannel.class);
 
-    private static final Comparator<SortableChannel> BY_SCORE = Comparator.comparingInt(SortableChannel::getScore);
-    private static final Duration FAILURE_MEMORY = Duration.ofSeconds(30);
-    private static final double FAILURE_WEIGHT = 10;
-
-    /**
-     * RTT_WEIGHT determines how sticky we are to the physically nearest node (as measured by RTT). If this is set
-     * too high, then we may deliver suboptimal perf by sending all requests to a slow node that is physically nearby,
-     * when there's actually a faster one further away.
-     * If this is too low, then we may prematurely spill across AZs and pay the $ cost. Keep this lower than
-     * {@link #FAILURE_WEIGHT} to ensure that a single 5xx makes a nearby node less attractive than a faraway node
-     * that exhibited zero failures.
-     */
-    private static final double RTT_WEIGHT = 3;
-
-    private final ImmutableList<MutableChannelWithStats> channels;
-    private final Random random;
-    private final Ticker clock;
-
-    @Nullable
-    private final RttSampler rttSampler;
+    private final BalancedScoreTracker tracker;
+    private final ImmutableList<LimitedChannel> channels;
 
     BalancedNodeSelectionStrategyChannel(
             ImmutableList<LimitedChannel> channels,
             Random random,
             Ticker ticker,
             TaggedMetricRegistry taggedMetrics,
             String channelName,
-            RttSampling samplingEnabled) {
+            RttSampling _samplingEnabled) {
         Preconditions.checkState(channels.size() >= 2, "At least two channels required");
-        this.random = random;
-        this.clock = ticker;
-        this.rttSampler = samplingEnabled == RttSampling.DEFAULT_OFF ? null : new RttSampler(channels, clock);
```

Comment:
I'd prefer to remove the RttSampler and related tests entirely, git won't forget about them.